src/toil_scripts/adam_uberscript/adam_uberscript.py

```diff
@@ -465,10 +475,29 @@ def collect_realtime_metrics(params, threshold=0.5):
                     try:
                         with cluster_size_lock:
                             cluster_size = get_cluster_size(params.cluster_name)
-                            if cluster_size > get_desired_cluster_size(domain):
-                                log.info('Terminating Instance: {}'.format(instance_id))
-                                log.info('Killing instance {0}\n'.format(instance_id))
-                                conn.terminate_instances(instance_ids=[instance_id])
+                            desired_cluster_size = get_desired_cluster_size(domain)
+                            if cluster_size > desired_cluster_size:
+                                log.info('Cluster size (%d) is larger than requested (%d).'
+                                         'Terminating idle instance %s.',
+                                         cluster_size,
+                                         desired_cluster_size,
+                                         instance_id)
+
+                                terminated_instances = conn.terminate_instances(instance_ids=[instance_id])
+                                assert len(terminated_instances) == 1, 'Terminating instance %s failed.' % instance_id
+
+                                # it is not clear as to whether boto's terminate call is blocking
+                                # so let's spin for a bit, because if we don't block on the
+                                # instance termination, we get a FUBAR synchronization bug
+                                cur_size = get_cluster_size(params.cluster_name)
+                                while cur_size >= cluster_size:
```

Comment:
Agreed. However, I don't think that the überscript properly handles spot termination right now anyways (see `ClusterSize` class), and I'm only running on in-demand clusters due to #126, and anyways we don't _actually_ support spot instances right now anyways (see #174).